### PR TITLE
feat: Upgrade hardhat to 2.5 to be compatible with London hardfork

### PR DESCRIPTION
**Pull request by nicholaspai**
Signed-off-by: Nick Pai <npai.nyc@gmail.com>

**Motivation**

[Hardhat 2.5 release notes](https://github.com/nomiclabs/hardhat/releases/tag/hardhat-core-v2.5.0). This PR updates the `hardhat` local test network to use the "london" hardfork and increases confidence that tested contracts will be compatible with the upcoming hardfork.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [X]  Existing tests adequate, no new tests required
- [X]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

**Review by chrismaree — APPROVED** [review]

File: packages/common/src/HardhatConfig.js [hunk 1]
```diff
@@ -44,7 +44,15 @@ function getHardhatConfig(configOverrides, workingDir = "./", includeTruffle = t
     },
     ovm: { solcVersion: "0.8.4-broken_alpha" },
     networks: {
-      hardhat: { gas: 11500000, blockGasLimit: 11500000, allowUnlimitedContractSize: false, timeout: 1800000 },
+      hardhat: {
+        hardfork: "london",
+        gasPrice: "auto",
+        initialBaseFeePerGas: 1_000_000_000,
+        gas: 11500000,
+        blockGasLimit: 11500000,
```
Comment (chrismaree) [inline packages/common/src/HardhatConfig.js:52]: is the limit not 15m?

File: packages/common/src/HardhatConfig.js [hunk 2]
```diff
@@ -44,7 +44,15 @@ function getHardhatConfig(configOverrides, workingDir = "./", includeTruffle = t
...
+        allowUnlimitedContractSize: false,
```
Comment (chrismaree) [inline packages/common/src/HardhatConfig.js:53]: is this not the default?

File: packages/common/src/HardhatConfig.js [hunk 3]
```diff
@@ -44,7 +44,15 @@ function getHardhatConfig(configOverrides, workingDir = "./", includeTruffle = t
     },
     ovm: { solcVersion: "0.8.4-broken_alpha" },
     networks: {
-      hardhat: { gas: 11500000, blockGasLimit: 11500000, allowUnlimitedContractSize: false, timeout: 1800000 },
+      hardhat: {
+        hardfork: "london",
```
Comment (mrice32) [inline packages/common/src/HardhatConfig.js:48]: So just so I understand, we're going to begin testing against london once this PR goes in?

File: packages/common/src/HardhatConfig.js [hunk 4]
```diff
@@ -44,7 +44,15 @@ function getHardhatConfig(configOverrides, workingDir = "./", includeTruffle = t
...
+        gasPrice: "auto",
```
Comment (mrice32) [inline packages/common/src/HardhatConfig.js:49]: This is pretty cool -- might be nice to use this on other networks too so we don't need to use our gas estimator in scripts!

**Review by mrice32 — APPROVED** [review]
LGTM!

File: packages/common/src/HardhatConfig.js (hunk 1 above)
Comment (nicholaspai) [reply packages/common/src/HardhatConfig.js:52]: Oh yes it increased, I'll change that

File: packages/common/src/HardhatConfig.js (hunk 2 above)
Comment (nicholaspai) [reply packages/common/src/HardhatConfig.js:53]: yeah it is i'll remove

File: packages/common/src/HardhatConfig.js (hunk 3 above)
Comment (nicholaspai) [reply packages/common/src/HardhatConfig.js:48]: correct, do you want to wait until mainnet switches over to merge this?

**Pull request merged**